pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -348,22 +347,18 @@ func (plugin *kubenetNetworkPlugin) setup(namespace string, name string, id kube
 		return err
 	}
 	// Coerce the CNI result version
-	res, err := cnitypes020.GetResult(resT)
+	res, err := cnitypescurrent.GetResult(resT)
 	if err != nil {
 		return fmt.Errorf("unable to understand network config: %v", err)
 	}
-	//TODO: v1.16 (khenidak) update NET_CONFIG_TEMPLATE to CNI version 0.3.0 or later so
-	// that we get multiple IP addresses in the returned Result structure
-	if res.IP4 != nil {
-		ipv4 = res.IP4.IP.IP.To4()
-	}
 
-	if res.IP6 != nil {
-		ipv6 = res.IP6.IP.IP
+	for _, ip := range res.IPs {
+		// Add the IP to tracked IPs
+		plugin.addPodIP(id, ip.Address.IP.String())
 	}
 
-	if ipv4 == nil && ipv6 == nil {
-		return fmt.Errorf("cni didn't report ipv4 ipv6")
+	if res.IPs == nil || len(res.IPs) == 0 {
```

Comment:
should this check go before the for loop?
 if there are no res.IPs we don't have to iterate over res.IPs

Reply:
Good catch ! Do we actually need the ` len(res.IPs) == 0` check ? Theoretically we should always get at least one result (loopback).

Reply:
@johscheuer you don't need the `nil` check; just `len(res.IPs) == 0` is fine.

Reply:
Changed it, thanks!

Reply:
check for res.IPs == nil is not necessary, cause len(nil) is 0. 
Got it, you put changes to separate commit "Correct check if CNI result has IPs", why not to amend a commit?